File: src/main/java/co/uk/jiveelection/campaign/output/twitter/TranslationEntity.java
package co.uk.jiveelection.campaign.output.twitter;

import com.google.auto.value.AutoValue;

@AutoValue
public abstract class TranslationEntity {
    public static TranslationEntity translate(int start, int end, String text) {
        return new AutoValue_TranslationEntity(start, end, text, true);
    }

    public static TranslationEntity verbatim(int start, int end, String text) {
        return new AutoValue_TranslationEntity(start, end, text, false);
    }

    public abstract int start();

    public abstract int end();

    public abstract String text();

    public abstract boolean translatable();
}

File: src/main/kotlin/co/uk/jiveelection/campaign/output/twitter/TranslationEntity.kt
package co.uk.jiveelection.campaign.output.twitter

/**
 * Describes information about a TwitterEntity.
 */
data class TranslationEntity(val start: Int, val end: Int, val text: String, val translatable: Boolean) {

    companion object {
        fun translate(start: Int, end: Int, text: String): TranslationEntity {
            return TranslationEntity(start, end, text, true)
        }

        fun verbatim(start: Int, end: Int, text: String): TranslationEntity {
            return TranslationEntity(start, end, text, false)
        }
    }
}
